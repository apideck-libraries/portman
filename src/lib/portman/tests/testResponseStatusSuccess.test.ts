import { OasMappedOperation, PostmanMappedOperation } from '../../'
import { getOasMappedOperation } from '../../../../__tests__/getOasMappedOperation'
import { getPostmanMappedOperation } from '../../../../__tests__/getPostmanMappedOperation'
import { testResponseStatusSuccess } from './testResponseStatusSuccess'

describe('testResponseStatusSuccess', () => {
  let oasOperation: OasMappedOperation
  let pmOperation: PostmanMappedOperation

  beforeEach(async () => {
    oasOperation = await getOasMappedOperation()
    pmOperation = await getPostmanMappedOperation()
  })

  it('should add test for status success', async () => {
    pmOperation = testResponseStatusSuccess(pmOperation, oasOperation)
    const pmTest = pmOperation.getTests()
    expect(pmTest.script.exec).toMatchSnapshot()
  })
})
