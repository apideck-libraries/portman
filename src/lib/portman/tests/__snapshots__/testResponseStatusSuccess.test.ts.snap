// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testResponseStatusSuccess should add test for status success 1`] = `
Array [
  "// Validate status 2xx 
pm.test(\\"[GET]::/crm/companies/:id - Status code is 2xx\\", function () {
   pm.response.to.be.success;
});
",
]
`;
